server/src/routes/track.ts

```diff
@@ -0,0 +1,108 @@
+import { Router } from 'express';
+import { z } from 'zod';
+import {
+  getArtists,
+  getRankOfTrack,
+  getTracks,
+  getTrackListenedCount,
+  getTrackFirstAndLastListened,
+  bestPeriodOfTrack,
+  getTrackRecentHistory,
+} from '../database';
+import { getAlbums } from '../database/queries/album';
+import { logger } from '../tools/logger';
+import { isLoggedOrGuest, validating } from '../tools/middleware';
+import { LoggedRequest, TypedPayload } from '../tools/types';
+
+const router = Router();
+export default router;
+
+const getTracksSchema = z.object({
+  ids: z.string(),
+});
+
+router.get(
+  '/:ids',
+  validating(getTracksSchema, 'params'),
+  isLoggedOrGuest,
+  async (req, res) => {
+    try {
+      const { ids } = req.params as TypedPayload<typeof getTracksSchema>;
+      const tracks = await getTracks(ids.split(','));
+      if (!tracks || tracks.length === 0) {
+        return res.status(404).end();
+      }
+      return res.status(200).send(tracks);
+    } catch (e) {
+      logger.error(e);
+      return res.status(500).end();
+    }
+  },
+);
+
+const getTrackStats = z.object({
+  id: z.string(),
+});
+
+router.get(
+  '/:id/stats',
+  validating(getTrackStats, 'params'),
+  isLoggedOrGuest,
+  async (req, res) => {
+    try {
+      const { user } = req as LoggedRequest;
+      const { id } = req.params as TypedPayload<typeof getTrackStats>;
+      const [ track ]= await getTracks([id]);
+      if (!track) {
+        return res.status(200).send({
+          code: 'NEVER_LISTENED',
+        });
+      }
```

Comment:
this is not consistent, the route below returns 404 if it does not find the song, here you return 200. For the artist, the NEVER_LISTENED code is returned if the total number of listens is equal to 0